homeassistant/components/fing/config_flow.py

```diff
@@ -0,0 +1,133 @@
+"""Config flow file."""
+
+import logging
+from typing import Any
+
+from fing_agent_api import FingAgent
+import httpx
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+
+from .const import AGENT_IP, AGENT_KEY, AGENT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_data_schema(
+    hass: HomeAssistant, user_input: dict[str, Any] | None = None
+) -> vol.Schema:
+    """Get a schema with default values."""
+
+    if user_input is None:
+        return vol.Schema(
+            {
+                vol.Required(CONF_NAME, default="Fing Agent"): str,
+                vol.Required(AGENT_IP): str,
+                vol.Required(AGENT_PORT, default="49090"): str,
+                vol.Required(AGENT_KEY): str,
+            }
+        )
+
+    return vol.Schema(
+        {
+            vol.Required(CONF_NAME, default=user_input.get(CONF_NAME)): str,
```

Comment:
Only helpers are allowed to have a user settable name. Other integrations just have to set it to something user recognizable